calcom/cal.com
# Added check on create eventtype to see is user has slug that already exists

added error check on onError

## What does this PR do?

If a user is creating a new event type and they provide a url slug that already exists an error message is displayed. 

Fixes #2701


## Type of change


## How should this be tested?

Create a new event type with URL '15min'
Create another event type with URL '15min'
Error Message should appear with Message: 'BAD_REQUEST:  URL already exists.'


## Checklist

- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
